apps/utils/locust_tests/common/bfd_user_base.py

```diff
@@ -164,7 +164,7 @@ def ssl_context_factory(self) -> ssl.SSLContext:
             context.load_verify_locations(cafile=self.server_public_key)
         return context
 
-    def get_by_url(self, url: str, headers: Optional[Dict[str, str]] = None, name: str = ""):
+    def get_by_url(self, url: str, headers: Optional[Mapping[str, str]] = None, name: str = ""):
```

Comment:
[`Mapping`](https://docs.python.org/3/library/collections.abc.html#collections.abc.Mapping) indicates that the argument is an immutable/read-only key-to-value collection.

It's used here to replace `dict` type hints for arguments that are never mutated. This replacement also fixed a few type errors.